posthog/hogql_queries/web_analytics/web_overview_pre_aggregated.py

```diff
@@ -0,0 +1,142 @@
+from typing import TYPE_CHECKING
+from datetime import datetime, UTC
+
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_select
+
+if TYPE_CHECKING:
+    from posthog.hogql_queries.web_analytics.web_overview import WebOverviewQueryRunner
+
+
+class WebOverviewPreAggregatedQueryBuilder:
+    # Supported property keys and their mapping to pre-aggregated table fields
+    SUPPORTED_PROPERTIES = {
+        "$host": "host",
+        "$device_type": "device_type",
+    }
+
+    def __init__(self, runner: "WebOverviewQueryRunner") -> None:
+        self.runner = runner
+
+    def can_use_preaggregated_tables(self) -> bool:
+        query = self.runner.query
+
+        # Only works for properties we know are in the pre-aggregated tables
+        for prop in query.properties:
+            if hasattr(prop, "key") and prop.key not in self.SUPPORTED_PROPERTIES:
+                return False
+
+        if query.conversionGoal:
+            return False
+
+        # Only work for fixed-dates that don't include current-date in the filters.
```

Comment:
Oh, that's very restrictive. Can't we just run the query for today on-the-fly?

Reply:
We can, but we would need to combine the results and they would be wrong for unique users, sessions average, bounce, etc... 

It will be possible, though; this was just the most MVP way of allowing them in prod. We will be able to use this hogql transform and combine today's data with the pre-aggregated one: https://github.com/PostHog/posthog/pull/32133